src/hypertable.c

```diff
@@ -347,7 +349,7 @@ static bool
 hypertable_is_compressed_or_materialization(Hypertable *ht)
 {
 	ContinuousAggHypertableStatus status = ts_continuous_agg_hypertable_status(ht->fd.id);
-	return (ht->fd.compressed || status == HypertableIsMaterialization);
+	return (TS_HYPERTABLE_INTERNAL_COMPRESSION(ht) || status == HypertableIsMaterialization);
```

Comment:
FWIW, this is also a readability improvement. Previously, it is too easy to believe that compressed=true means a regular hypertable with compression enabled.